File: final/badstrand/data/gen.cpp
#include <bits/stdc++.h>

#define rep(i, a, b) for (int i = a; i < int(b); ++i)
#define all(v) v.begin(), v.end()
#define trav(a, v) for (auto &a : v)

typedef long long ll;
typedef long double ld;

const long long inf = 2e9;

using namespace std;
vector<string> args;
string cmdlinearg(string name)
{
    trav(a, args)
    {
        if (a.size() < name.size() + 1)
            continue;
        bool good = 1;
        rep(i, 0, name.size())
        {
            if (a[i] != name[i])
            {
                good = 0;
                break;
            }
            if (a[name.size()] != '=')
                good = 0;
        }
        if (good)
        {
            string res = "";
            rep(i, name.size() + 1, a.size()) res += a[i];
            return res;
        }
    }
    cout << "paremeter " << name << " not found" << endl;
    assert(0);
}
ll convert_to_int(string s)
{
    ll val = 0;
    reverse(all(s));
    ll power = 1;
    rep(i, 0, s.size()) val += power * (s[i] - '0'), power *= 10;
    return val;
}
int main(int argc, char **argv)
{
    // The areguments should be in this format:
    // seed=(seed) max_k=(max_k) max_n=(max_n)
    rep(i, 0, argc) args.push_back(argv[i]);
    ll seed = convert_to_int(args[argc-1]);
    srand(seed);
    ll max_a = convert_to_int(cmdlinearg("max_a"));
    ll n = convert_to_int(cmdlinearg("max_n"));
    ll ans = convert_to_int(cmdlinearg("ans"));

    vector<ll> as;
    rep(i,0,n) as.push_back(rand()%1000 + 1);

    set<ll> s;
    rep(i,0,ans) s.insert(a[i]);
    ll b = *s.begin();
    rep(i,0,)

    cout << n << " " << rand() % (max_k + 1) << endl;
    rep(i, 0, n)
    {
        cout << rand() % 360 << (i + 1 == n ? "" : " ");
    }
    cout << endl;
    return 0;
}

File: final/badstrand/data/gen.py
import sys
import random

def cmdlinearg(name, default=None):
    for arg in sys.argv:
        if arg.startswith(name + "="):
            return arg.split("=")[1]
    assert default is not None, name
    return default

random.seed(int(cmdlinearg('seed', sys.argv[-1])))
n = int(cmdlinearg('n'))
ans = int(cmdlinearg('ans'))
mode = cmdlinearg('mode',"none")

a = [random.randint(1,1000) for i in range(n)]

if mode=="same_a":
    for i in range(n):
        a[i] = a[0]

s = sum(a[:ans])
b = s
for i in range(n-ans):
    s += a[i+ans] - a[i]
    b = min(b,s)
print(n,b)
print(" ".join(str(x) for x in a))